caffe2/python/pybind_state.cc

```diff
@@ -1107,6 +1107,26 @@ void addGlobalMethods(py::module& m) {
         const auto c = schema->InferCost(def, shapes);
         return std::make_tuple(c.flops, c.bytes_moved);
       });
+  m.def(
+      "get_operator_input_blob_shape",
+      [](const py::bytes& op_def, const std::vector<string>& input_blobs) {
+        CAFFE_ENFORCE(gWorkspace);
+        OperatorDef def;
+        CAFFE_ENFORCE(
+            ParseProtoFromLargeString(op_def.cast<std::string>(), &def),
+            "Couldn't parse operator proto.");
+        auto* schema = OpSchemaRegistry::Schema(def.type());
+        CAFFE_ENFORCE(schema);
+        std::unordered_map<string, std::vector<int64_t>> shapes;
+        for (const auto& blob_name : input_blobs) {
+          auto* blob = gWorkspace->GetBlob(blob_name);
+          auto ts = GetTensorShapeOfBlob(blob);
```

Comment:
`GetTensorShapeOfBlob` constructs a TensorShape struct. Shouldn't we be able to directly to get the shape of the Tensor in c++? (of course we need to check whether the blob is of type Tensor first).